src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -79,6 +98,29 @@ def extract_strings(self) -> Mapping[int, Set[str]]:
                 )
                 continue
 
+    @metrics.wraps("process_messages.limit_messages")
+    def limit_messages(self, keys_to_remove: Sequence[PartitionIdxOffset]) -> None:
+        for offset in keys_to_remove:
+            metrics.incr(
+                "sentry_metrics.indexer.process_messages.dropped_message",
+                tags={
+                    "reason": "cardinality_limit",
+                },
+            )
+            if _should_sample_debug_log():
+                logger.error(
+                    "process_messages.dropped_message",
+                    extra={
+                        "reason": "cardinality_limit",
+                    },
+                )
+
+            self.skipped_offsets.add(offset)
+
+    @metrics.wraps("process_messages.extract_strings")
+    def extract_strings(self) -> Mapping[int, Set[str]]:
+        org_strings = defaultdict(set)
+
         for partition_offset, message in self.parsed_payloads_by_offset.items():
```

Comment:
I think you also need to skip the messages that you will have removed via `limit_messages` (or filter if we rename it). 
Otherwise we would be returning strings that we filtered through the cardinality limiter.

Reply:
fixed